res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,978 @@
+    this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel1]_Effect1]", "parameter1", 0x11, 0xFFFF, this.parameterHandler);
+
+    this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter3", 0x1F, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter2", 0x21, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter1", 0x23, 0xFFFF, this.parameterHandler);
+
+    this.registerInputScaler(messageLong, "[QuickEffectRack1_[Channel1]]", "super1", 0x13, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[QuickEffectRack1_[Channel2]]", "super1", 0x25, 0xFFFF, this.parameterHandler);
+
+    this.registerInputScaler(messageLong, "[Master]", "crossfader", 0x05, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[Master]", "gain", 0x15, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[Sampler]", "pregain", 0x17, 0xFFFF, this.samplerPregainHandler);
+    this.registerInputScaler(messageLong, "[Master]", "headMix", 0x19, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[Master]", "headGain", 0x1B, 0xFFFF, this.parameterHandler);
+
+    this.controller.registerInputPacket(messageLong);
+
+    // Soft takeover for all knobs
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+
+    engine.softTakeover("[Channel1]", "volume", true);
+    engine.softTakeover("[Channel2]", "volume", true);
+
+    engine.softTakeover("[Channel1]", "pregain", true);
+    engine.softTakeover("[Channel2]", "pregain", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[QuickEffectRack1_[Channel1]]", "super1", true);
+    engine.softTakeover("[QuickEffectRack1_[Channel2]]", "super1", true);
+
+    engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "gain", true);
+    engine.softTakeover("[Master]", "headMix", true);
+    engine.softTakeover("[Master]", "headGain", true);
+
+    for (var i = 1; i <= 16; ++i) {
+        engine.softTakeover("[Sampler" + i + "]", "pregain", true);
+    }
+
+    // Dirty hack to set initial values in the packet parser
+    var data = TraktorS2MK3.toBytes("01 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 00 00 00 00");
```

Comment:
The `toBytes` function seems like an overcomplication to me. Why not just write out the array literal here?

I'm also not clear why you need to pass this initial packet to `incomingData`

Reply:
I removed this debug function now and use an array instead in https://github.com/mi01/mixxx/commit/3f0fd915ae129993968be79bcb867eb8f22bb468

Passing this initial values to the HID packet parser solve an issue i have. The parser seems to react only to "changes" of its cached state. Without this line the first button press after startup is not "recognized" and no callback function is triggered, only the second button press is handled right.